packages/powersync_attachments_stream/lib/src/storage/io_local_storage.dart

```diff
@@ -0,0 +1,127 @@
+/// Local storage adapter for handling file operations on the device filesystem.
+///
+/// This file implements the [IOLocalStorage] class, which provides methods for
+/// saving, reading, deleting, copying, and managing files and directories using
+/// the Dart IO library. It is used as the local storage backend for attachments
+/// in the PowerSync attachments system.
+///
+/// Features:
+/// - Save files from streams (creates directories and all necessary parents dynamically if they do not exist)
+/// - Read files as streams
+/// - Delete files and their metadata
+/// - Copy files and their metadata
+/// - Create and remove directories (creates all necessary parents dynamically)
+/// - Check file existence
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+import 'package:path/path.dart' as p;
+import '../abstractions/local_storage.dart';
```

Comment:
nit: Typically one inserts an empty line between `dart:` and `package:` imports and then another one before local imports. All the [effective Dart samples](https://dart.dev/effective-dart/style#do-place-dart-imports-before-other-imports) do that, but it's not spelled out explicitly. I've added a suggestion for this here, it affects other files as well.

```suggestion
import 'dart:async';
import 'dart:io';
import 'dart:typed_data';

import 'package:path/path.dart' as p;

import '../abstractions/local_storage.dart';
```